reddit_service_ads_tracking/lib/events.py

```diff
@@ -0,0 +1,137 @@
+import baseplate.events
+import logging
+
+from urllib import unquote
+from urlparse import urlparse
+
+
+from . import useragent
+
+logger = logging.getLogger(__name__)
+
+
+def _get_domain(url):
+    return urlparse(url).netloc
+
+
+class EventQueue(baseplate.events.EventQueue):
+    def click_event(self, url,
```

Comment:
I know having these functions that generate events and call `self.save_event` is the pattern in r2, but is there any reason to replicate it here? Putting events into the queue and stuffing them with data are two different things, so it's a violation of the single responsibility principle.